visualization/src/interactor_style.cpp

```diff
@@ -162,13 +162,22 @@ pcl::visualization::PCLVisualizerInteractorStyle::saveCameraParameters (const st
 
 //////////////////////////////////////////////////////////////////////////////////////////////
 void
-pcl::visualization::PCLVisualizerInteractorStyle::getCameraParameters (pcl::visualization::Camera &camera)
+pcl::visualization::PCLVisualizerInteractorStyle::getCameraParameters (pcl::visualization::Camera &camera, int viewport) const
 {
-  FindPokedRenderer (Interactor->GetEventPosition ()[0], Interactor->GetEventPosition ()[1]);
-
-  auto window = Interactor->GetRenderWindow ();
-  auto cam = window->GetRenderers ()->GetFirstRenderer ()->GetActiveCamera ();
-  camera = Camera (*cam, *window);
+  rens_->InitTraversal ();
+  vtkRenderer* renderer = NULL;
+  int i = 0;
+  while ((renderer = rens_->GetNextItem ()) != NULL)
+  {
+    if (viewport == 0 || viewport == i)
```

Comment:
This check can simply be `viewport == i`, because you break out of the while loop the first time the check is true. Therefore that behaviour of `viewport == 0` having the underlying meaning that it should traverse all viewports, has no effect here.

Reply:
Good point!